cmake/onnxruntime_providers.cmake

```diff
@@ -633,7 +633,7 @@ if (onnxruntime_USE_OPENVINO)
       else()
         list(APPEND OPENVINO_LIB_DIR_LIST $ENV{INTEL_OPENVINO_DIR}/deployment_tools/inference_engine/lib/intel64/Release)
       endif()
-    elseif(CMAKE_LIBRARY_ARCHITECTURE STREQUAL "aarch64-linux-gnu")
+    elseif(CMAKE_SYSTEM_PROCESSOR STREQUAL "aarch64")
```

Comment:
ok for now. 
but this may need to be restructured to make clear what the different cases are.
i think currently it's implied that the elseif and else cases are Linux arm64 and Linux x86. btw OpenVINO supports MacOS as well. not sure if that works currently.